src/libraries/System.Private.CoreLib/src/System/Buffer.cs

```diff
@@ -341,7 +332,7 @@ internal static void Memmove<T>(ref T destination, ref T source, nuint elementCo
         private static void Memmove(ref byte dest, ref byte src, nuint len)
         {
             // P/Invoke into the native version when the buffers are overlapping.
-            if (((nuint)Unsafe.ByteOffset(ref src, ref dest) < len) || ((nuint)Unsafe.ByteOffset(ref dest, ref src) < len))
+            if (((nuint)(nint)Unsafe.ByteOffset(ref src, ref dest) < len) || ((nuint)(nint)Unsafe.ByteOffset(ref dest, ref src) < len))
```

Comment:
@cston, `Unsafe.ByteOffset` returns `IntPtr` and it looks like there isn't an allowed explicit conversion from `IntPtr` to `nuint`, is that by design?

Reply:
The compiler should allow explicit conversion from `IntPtr` to `nuint`. [https://github.com/dotnet/roslyn/issues/42560](https://github.com/dotnet/roslyn/issues/42560)

---
In reply to: [422452137](https://github.com/dotnet/runtime/pull/36159#discussion_r422452137) [](ancestors = 422452137)